pkg/hooks/manager.go

```diff
@@ -0,0 +1,202 @@
+package hooks
+
+import (
+	"context"
+	"encoding/json"
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"google.golang.org/grpc"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	hooksInfo "kubevirt.io/kubevirt/pkg/hooks/info"
+	hooksV1alpha1 "kubevirt.io/kubevirt/pkg/hooks/v1alpha1"
+	"kubevirt.io/kubevirt/pkg/log"
+	domainSchema "kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+	virtwrapApi "kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+type callackClient struct {
+	SocketPath          string
+	Version             string
+	subsribedHookPoints []*hooksInfo.HookPoint
+}
+
+var manager *Manager
+var once sync.Once
+
+type Manager struct {
+	collected             bool
+	callbacksPerHookPoint map[string][]*callackClient
+}
+
+func GetManager() *Manager {
```

Comment:
I looked this file over a few times trying to determine if there are any concurrency issues with this globally accessible manager object. I didn't find any.  Maps are safe to read from concurrently as long as they are not written to at the same time. https://golang.org/doc/faq#atomic_maps

Since we populate the map upfront, we should be fine here.